docs/Troubleshooting.md

```diff
@@ -21,3 +21,23 @@ You can kill the process by:
 
   * on Windows OS: run `taskkill /F /PID ...` all instances
   * on other OS: run `kill -9 ...` all instances
+
+### The Language Server Crashes Due to an Out Of Memory Error
+
+If you are working with large XML files or referencing large schema files,
+it's likely that increasing the memory available to the language server will resolve this issue.
+
+If you aren't working with large XML files, then it may be a memory leak.
+Please [file a issue](https://github.com/redhat-developer/vscode-xml/issues) with a description of what you were doing if this is the case.
+
+#### How to increase the amount of memory available to the Language Server
+
+If you are using the Java Language Server:
+1. Go to settings
+2. Navigate to the setting `xml.server.vmargs`
+3. Add `-Xmx 1G` to the setting string. This allows the the language server to use at most 1 gigabyte of memory.
+4. If the problem persists, you can increase the `1G` to `2G` or higher
+
+If you are using the binary language server:

```

Comment:
I probably wouldn't format the binary language server part as "steps" since this is more informational. I would maybe mention that OOM issues are less likely to occur with the binary language server. Maybe we could mention that those running into memory issues should be encouraged to try the binary server as a workaround ?